File: java/8 - Utilities/Ternary Search.java
Retorna el valor minimo de una funcion entre l y r. Se recomienda usar 50 iteraciones.

static double f(double x) {
    return //funcion a evaluar que depende de x
}

static double ternary_search(double l, double r, int it) {
    double a = (2.0*l + r)/3.0;
    double b = (l + 2.0*r)/3.0;
    if (it == 0) return f(a);
    if (f(a) < f(b)) return ternary_search(l, b, it-1);
    return ternary_search(a, r, it-1);
}

File: c++/Math/Ternary Search.cpp
Retorna el valor máximo de una función f(x) entre [l, r]

const double eps = 1e-9;

double ternary_search(double l, double r) {
    while (r - l > eps) { // O se pueden hacer unas 100 iteraciones
        double m1 = l + (r - l) / 3;
        double m2 = r - (r - l) / 3;
        double f1 = f(m1);
        double f2 = f(m2);
        if (f1 < f2) l = m1;
        else r = m2;
    }
    return f(l);
}
